oshi-core/src/main/java/oshi/hardware/GlobalMemory.java

```diff
@@ -1,58 +1,67 @@
- * physical memory (RAM) as well as any available virtual memory.
- */
-public interface GlobalMemory {
-    /**
-     * The amount of actual physical memory, in bytes.
-     *
-     * @return Total number of bytes.
-     */
-    long getTotal();
-
-    /**
-     * The amount of physical memory currently available, in bytes.
-     *
-     * @return Available number of bytes.
-     */
-    long getAvailable();
-
-    /**
-     * The number of bytes in a memory page
-     *
-     * @return Page size in bytes.
-     */
-    long getPageSize();
-
-    /**
-     * Virtual memory, such as a swap file.
-     *
-     * @return A VirtualMemory object.
-     */
-    VirtualMemory getVirtualMemory();
-}
+/**
+ * OSHI (https://github.com/oshi/oshi)
+ *
+ * Copyright (c) 2010 - 2019 The OSHI Project Team:
+ * https://github.com/oshi/oshi/graphs/contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package oshi.hardware;
+
+import java.util.List;
+
+/**
+ * The GlobalMemory class tracks information about the use of a computer's
+ * physical memory (RAM) as well as any available virtual memory.
+ */
+public interface GlobalMemory {
+    /**
+     * The amount of actual physical memory, in bytes.
+     *
+     * @return Total number of bytes.
+     */
+    long getTotal();
+
+    /**
+     * The amount of physical memory currently available, in bytes.
+     *
+     * @return Available number of bytes.
+     */
+    long getAvailable();
+
+    /**
+     * The number of bytes in a memory page
+     *
+     * @return Page size in bytes.
+     */
+    long getPageSize();
+
+    /**
+     * Virtual memory, such as a swap file.
+     *
+     * @return A VirtualMemory object.
+     */
+    VirtualMemory getVirtualMemory();
+    
+    /**
+     * Physical memory, such as banks of memory.
+     *
+     * @return A list of PhysicalMemory objects.
+     */
+    List<PhysicalMemory> getPhysicalMemory();

```

Comment:
See #757 where I initially thought to change all of OSHI's "collection" type API to Lists, and ended up not doing it due to serialization issues.  That said, I've since decided to get rid of serialization so that's not a barrier.  Do Lists work with Jackson's ObjectMapper?

Even then, for consistency, I think I'd prefer to return a fixed array here, and consider a List in the next major release.

Reply:
Changed from list to array and set size to number of banks